shared/util/saga.js

```diff
@@ -125,25 +125,62 @@ function* sequentially(effects: Array<any>): Generator<any, Array<any>, any> {
 // whatever purework returns will be yielded on.
 // i.e. it can return put(someAction). That effectively transforms the input action into another action
 // It can also return all([put(action1), put(action2)]) to dispatch multiple actions
-function safeTakeEveryPure<A, R, FinalAction>(
+function safeTakeEveryPure<A, R, FinalAction, FinalActionError>(
   pattern: string | Array<any> | Function,
   pureWorker: ((action: A, state: TypedState) => any) | ((action: A) => any),
-  actionCreatorsWithResult?: (result: R) => FinalAction
+  actionCreatorsWithResult?: (result: R) => FinalAction,
+  actionCreatorsWithError?: (result: R) => FinalActionError
```

Comment:
added a way to handle errors

Reply:
Do you really need this? I think you can do without it, and instead use the CallAndWrap.

Reply:
nvm, after thinking about it some more (especially in the call is cancelled case) I think it makes sense. 👍